gdsfactory/typings.py

```diff
@@ -45,6 +45,48 @@
 from gdsfactory.port import Port
 from gdsfactory.technology import LayerLevel, LayerStack
 
+
+STEP_DIRECTIVES = {
+    "x",
+    "y",
+    "dx",
+    "dy",
+}
+_STEP = Literal[
+    "x",
+    "y",
+    "dx",
+    "dy",
+]
+
+STEP_DIRECTIVES_ALL_ANGLE = {
+    "x",
+    "y",
+    "dx",
+    "dy",
+    "ds",
+    "exit_angle",
+    "cross_section",
+    "connector",
+    "separation",
+}
+
+_STEP_ALL_ANGLE = Literal[
+    "x",
+    "y",
+    "dx",
+    "dy",
+    "ds",
+    "exit_angle",
+    "cross_section",
+    "connector",
+    "separation",
```

Comment:
A key part of #1255 is documenting what these fields actually do. It's fine if this is addressed in a separate PR, just wanted to highlight the issue.

Reply:
@tvt173 

Troy, can you help us document this?

Reply:
see

https://github.com/gdsfactory/gdsfactory/pull/1257/commits/0a717c4b08807086126dc0468c6a73d290a4fb58